app/lib/screens/signing/sign_with_text.dart

```diff
@@ -0,0 +1,110 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:threebotlogin/providers/wallets_provider.dart';
+import 'package:threebotlogin/screens/signing/signing_mixin.dart';
+
+class SignWithTextScreen extends ConsumerStatefulWidget {
+  const SignWithTextScreen({super.key});
+
+  @override
+  ConsumerState<SignWithTextScreen> createState() => _SignWithTextScreenState();
+}
+
+class _SignWithTextScreenState extends ConsumerState<SignWithTextScreen>
+    with SigningMixin {
+  String? textError;
+
+  @override
+  void validateInputs() {
+    setState(() {
+      if (textController.text.isEmpty) {
+        textError = 'Please enter text to sign';
+      } else {
+        textError = null;
+      }
+    });
+    validateWallet();
+  }
+
+  @override
+  void setSigningError(String message) {
+    setState(() {
+      textError = message;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final wallets = ref.watch(walletsNotifier);
```

Comment:
What if the wallets were not loaded?

Reply:
https://github.com/user-attachments/assets/cae09358-58d0-4710-9ef6-df3cf13a126b